src/track/external-captions-handler.js

```diff
@@ -311,44 +311,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
   _handleCaptionOnTimeUpdate(track: TextTrack): void {
     const currentTime = this._player.currentTime;
     if (currentTime) {
-      if (this._activeTextCues.length > 0 && currentTime < this._activeTextCues[this._activeTextCues.length - 1].startTime) {
-        this._activeTextCues = [];
-        this.dispatchEvent(new FakeEvent(CustomEventType.TEXT_CUE_CHANGED, {cues: []}));
-      } else {
-        if (Math.abs(currentTime - this._lastTimeUpdate) > 1) {
-          this._maybeSetExternalCueIndex();
-        }
-        const cues = this._textTrackModel[track.language].cues;
-        let activeCuesChanged = false;
-        for (let activeTextCuesIndex = 0; activeTextCuesIndex < this._activeTextCues.length; activeTextCuesIndex++) {
-          const cue = this._activeTextCues[activeTextCuesIndex];
-          if (currentTime < cue.startTime || cue.endTime < currentTime) {
-            this._activeTextCues.splice(activeTextCuesIndex, 1);
-            activeCuesChanged = true;
-          }
-        }
-        while (this._externalCueIndex < cues.length && currentTime > cues[this._externalCueIndex].startTime) {
-          this._activeTextCues.push(cues[this._externalCueIndex]);
-          this._externalCueIndex++;
-          activeCuesChanged = true;
-        }
-        if (activeCuesChanged) {
-          this.dispatchEvent(new FakeEvent(CustomEventType.TEXT_CUE_CHANGED, {cues: this._activeTextCues}));
-        }
+      const isSeeked = this._isSeeked() && this._maybeSetExternalCueIndex();
```

Comment:
`cueIndexUpdated` sounds better and it will give you a better expression convention  (all are past tense finished operations):
```javascript
if (cueIndexUpdated || activeCuesAdded || activeCuesRemoved) {
```

Reply:
also `hadSeeked`